vendor/bat-native-ledger/src/bat/ledger/internal/bitflyer/bitflyer_authorization.cc

```diff
@@ -0,0 +1,196 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/bitflyer/bitflyer_authorization.h"
+
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "bat/ledger/global_constants.h"
+#include "bat/ledger/internal/bitflyer/bitflyer_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/logging/event_log_keys.h"
+#include "crypto/sha2.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+using std::placeholders::_4;
+
+namespace ledger {
+namespace bitflyer {
+
+BitflyerAuthorization::BitflyerAuthorization(LedgerImpl* ledger)
+    : ledger_(ledger),
+      bitflyer_server_(std::make_unique<endpoint::BitflyerServer>(ledger)) {}
+
+BitflyerAuthorization::~BitflyerAuthorization() = default;
+
+void BitflyerAuthorization::Authorize(
+    const base::flat_map<std::string, std::string>& args,
+    ledger::ExternalWalletAuthorizationCallback callback) {
+  const auto wallet = ledger_->wallet()->GetWallet();
+  if (!wallet) {
+    BLOG(0, "Wallet is null");
+    callback(type::Result::LEDGER_ERROR, {});
+    return;
+  }
+
+  auto bitflyer_wallet = GetWallet(ledger_);
+  if (!bitflyer_wallet) {
+    BLOG(0, "Wallet is null");
+    callback(type::Result::LEDGER_ERROR, {});
+    return;
+  }
+
+  const auto current_one_time = bitflyer_wallet->one_time_string;
```

Comment:
Is there a reason why we store this string in the wallet? As far as I can tell, this string can be generated for authorization and discarded once the authorization response is validated?

Reply:
Good point - we're taking the same approach here as our Uphold implementation. The oauth state in fact doesn't need to be persisted between browser sessions, so it could be stored in memory instead. After discussing with @emerick we'd like to improve this for both Uphold and bitFlyer with a future PR. Would that be OK?

Reply:
sgtm